### ISSUE TITLE
Value components. Match.

### ISSUE THREAD

**Issue by Praxie** [Pliohub/Plio#1296]
Acceptance criteria is the same as in #1295.

**Comment by sives** [comment]
User messages need to be improved, see #1295

**Comment by Dartv** [comment]
@sives I encountered an issue with the current design. If you have value component matched to customer segment AND that customer segment matched to the same value component and you also have some customer elements matched, then if you change `matched to` of one of value proposition or customer segment to "None", your matching references between customer elements will be reset but you will still be able to match them from the value component or customer segment that still has `matched to` pointing to the other document.

One solution would be changing `matched to` field of both value proposition and customer segment to "None" if they're matched.

**Comment by sives** [comment]
@Dartv Before answering this question, I want to clarify the relationships between the business objects, see below:

You can match a Value Proposition to a Customer Segment.
You can't directly match a Value Component to a Customer Segment.    
A Feature is a type of Value Component
A Benefit is a type of Value Component.
A Value Proposition contains Value Components (i.e. contains Features and Benefits)

You can match a Customer Segment to a Value Proposition
You can't match a Customer Insight to a Value Proposition
A Need is a type of Customer insight.
A Want is a type of Customer Insight.
A Customer Segment contains Customer Insights (i.e. contains Needs and Wants)

You can match a Customer Segment to a Value Proposition (It is important to understand that "Customer Segment A matched to Value Proposition X" is an identical matching relationship to "Value Proposition X matched to Customer Segment A".  

This means that in the Customer Segment A card, if you change the value in the "Matched to" field from "Value Proposition X" to "None", then the value in the "Matched to" field in the Value Proposition X card must automatically be changed from "Customer Segment A" to "None".

Hope this is clear :-)

**Comment by Dartv** [comment]
@sives but can one value proposition be matched to multiple customer segments and vice-versa?

**Comment by unplugged** [comment]
Yes

**Comment by sives** [comment]
@Dartv No, this would create a lot more complexity, so it got simplified back in June when I was discussing this with Mike. See the requirements specification, in the section titled "New Customer Insights and Value Components subcards":

"A single Customer Segment can only be linked to a single Value Proposition within the Customer Insights subcard section of the Customer Segment card.  Similarly, it a single Value Proposition can only be linked to a single Customer Segment within the Value Components subcard section of the Value Proposition card. (this replaces the requirement in the previous version of the spec to support one-to-multiple linkages)."

**Comment by Dartv** [comment]
https://github.com/Pliohub/Plio/pull/1408
https://github.com/Pliohub/Plio/pull/1407
https://github.com/Pliohub/Plio/pull/1404
https://github.com/Pliohub/Plio/pull/1401
https://github.com/Pliohub/Plio/pull/1400
https://github.com/Pliohub/Plio/pull/1399
